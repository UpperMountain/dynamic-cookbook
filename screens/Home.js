import React from "react";
import {
  View,
  StyleSheet,
  TouchableWithoutFeedback,
  Keyboard
} from "react-native";
import { LinearGradient } from "expo";
import SearchBar from "../components/SearchBar";
import { MealCard } from "../components/MealCard";
import { FontAwesome } from "@expo/vector-icons";
import tabIcon from "../components/tabIcon";

class Home extends React.Component {
  constructor(props) {
    super();
    this.state = {
      searching: false
    };
  }

  static navigationOptions = {
    tabBarIcon: tabIcon("home")
  };

  render() {
    const DismissKeys = ({ children }) => (
      <TouchableWithoutFeedback
        onPress={() => {
          Keyboard.dismiss();
        }}
      >
        {children}
      </TouchableWithoutFeedback>
    );

    return (
      <DismissKeys>
        <View style={styles.container}>
          <LinearGradient
            style={styles.background}
            colors={["#FFAFBD", "#FFC3A0"]}
          >
            <SearchBar open={this.state.searching} />
          </LinearGradient>
        </View>
      </DismissKeys>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "transparent"
  },
  background: {
    flex: 1,
    alignItems: "center"
  }
});

export default Home;
