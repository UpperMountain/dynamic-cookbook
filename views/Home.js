import React from "react";
import {
  View,
  StyleSheet,
  TouchableWithoutFeedback,
  Keyboard,
  StatusBar
} from "react-native";
import { LinearGradient } from "expo";
import MenuBar from "../components/MenuBar";
import SearchBar from "../components/SearchBar";
import { List } from "../components/RecipeList";
import { MealCard } from "../components/MealCard";

class Home extends React.Component {
  constructor(props) {
    super();
    this.state = {
      searching: false
    };
  }

  render() {
    const DismissKeys = ({ children }) => (
      <TouchableWithoutFeedback
        onPress={() => {
          Keyboard.dismiss();
        }}
      >
        {children}
      </TouchableWithoutFeedback>
    );

    return (
      <DismissKeys>
        <View style={styles.container}>
          <StatusBar barStyle="light-content" />
          <LinearGradient
            style={styles.background}
            colors={["#FFAFBD", "#FFC3A0"]}
          >
            <SearchBar open={this.state.searching} />
            <MenuBar />
          </LinearGradient>
        </View>
      </DismissKeys>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "transparent"
  },
  background: {
    flex: 1,
    alignItems: "center"
  }
});

export default Home;
