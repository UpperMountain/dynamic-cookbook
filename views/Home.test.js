import React from "react";
import Home from "./Home";
import renderer from "react-test-renderer";

it("renders without exploding", () => {
  const cmp = renderer.create(<Home />);
});

it("passes snapshot tests", () => {
  const cmp = renderer.create(<Home />);
  expect(cmp.toJSON()).toMatchSnapshot();
});
