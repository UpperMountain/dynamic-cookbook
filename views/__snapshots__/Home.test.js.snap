// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`passes snapshot tests 1`] = `
<View
  accessible={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "transparent",
      "flex": 1,
    }
  }
>
  <ExponentLinearGradient
    colors={
      Array [
        4294946749,
        4294951840,
      ]
    }
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(255, 255, 255, .2)",
          "borderRadius": 72,
          "display": "flex",
          "flexDirection": "row",
          "height": 56,
          "marginTop": 56,
          "paddingLeft": 36,
          "paddingRight": 36,
          "width": 315,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Search Recipes..."
        placeholderTextColor="white"
        style={
          Object {
            "color": "white",
            "flex": 1,
            "fontFamily": "raleway",
            "fontSize": 18,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <Text />
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#fff",
            "borderTopColor": "rgba(108, 123, 138, .24)",
            "borderTopWidth": 1,
            "bottom": 0,
            "flexDirection": "row",
            "height": 74,
            "justifyContent": "space-around",
            "position": "absolute",
            "width": "100%",
          },
          Object {
            "elevation": 5,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 5,
          },
        ]
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "marginTop": 5,
            "opacity": 1,
          }
        }
      >
        <Text />
        <Text
          style={
            Object {
              "color": "#ffc3a0",
            }
          }
        >
          Home
        </Text>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "marginTop": 5,
            "opacity": 1,
          }
        }
      >
        <Text />
        <Text
          style={
            Object {
              "color": "#757575",
            }
          }
        >
          Categories
        </Text>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "marginTop": 5,
            "opacity": 1,
          }
        }
      >
        <Text />
        <Text
          style={
            Object {
              "color": "#757575",
            }
          }
        >
          My Meal
        </Text>
      </View>
    </View>
  </ExponentLinearGradient>
</View>
`;
